### ISSUE TITLE
(=)/2 fatal error bug check

### ISSUE THREAD

**Issue by flexoron** [trealla-prolog/trealla#278]
v2.23.14
```
?- C = + +1.
   C = + +1.

?- B = +B, C = + +1, B == C.
   B = + + + + + + + + + + +  ..., C = B . % Error C = B , check bug B == C.
```
```
So, this guarded degenerated code is not portable!?
?- C = + +1, B = +B, C == B, X is C.
   C = + +1, B = C, X = 1.

SWI,GNU,ECLiPSe
?- C = + +1, B = +B, C == B, X is C.
false.
```

**Comment by flexoron** [comment]
I guess it is more than a printing issue
v2.23.16
```
?- C = + +1, B = +B, C == B, X is C.
   C = + +1, B = +B, X = 1.

SWI,GNU,ECLiPSe
?- C = + +1, B = +B, C == B, X is C.
false.

**Comment by flexoron** [comment]
v2.23.18
```
Trealla thinks C == B is true.

?- C = + +1, B = +B, C == B, X is C.
   C = + +1, B = + + + + + + + + + + + ..., X = 1.
```

**Comment by flexoron** [comment]
Fixed with v2.23.19
Crunching closing.
